controllers/renovate_util.go

```diff
@@ -22,63 +24,173 @@ import (
 	. "github.com/konflux-ci/build-service/pkg/common"
 	"github.com/konflux-ci/build-service/pkg/git"
 	"github.com/konflux-ci/build-service/pkg/git/github"
+	"github.com/konflux-ci/build-service/pkg/git/gitproviderfactory"
+	"github.com/konflux-ci/build-service/pkg/k8s"
 	"github.com/konflux-ci/build-service/pkg/logs"
 )
 
 const (
 	RenovateImageEnvName    = "RENOVATE_IMAGE"
 	DefaultRenovateImageUrl = "quay.io/redhat-appstudio/renovate:v37.74.1"
+	DefaultRenovateUser     = "redhat-hat-konflux"
 )
 
-type installationStruct struct {
-	id           int
-	component    string
-	token        string
-	repositories []renovateRepository
-}
-
 type renovateRepository struct {
 	Repository   string   `json:"repository"`
 	BaseBranches []string `json:"baseBranches,omitempty"`
 }
 
-// GetGithubInstallationsForComponents This method avoids iterating over all installations, it is intended to be called when the component list is small
-func GetGithubInstallationsForComponents(ctx context.Context, client client.Client, eventRecorder record.EventRecorder, componentList []v1alpha1.Component) (string, []installationStruct, error) {
+// UpdateTarget represents a target source code repository to be executed by Renovate with credentials and repositories
+type updateTarget struct {
+	ComponentName string
+	GitProvider   string
+	Username      string
+	GitAuthor     string
+	Token         string
+	Endpoint      string
+	Repositories  []renovateRepository
+}
+
+/*
+type ComponentDependenciesUpdater interface {
+	GetUpdateTargetsGithubApp(ctx context.Context, componentList []v1alpha1.Component) []updateTarget
+	GetUpdateTargetsBasicAuth(ctx context.Context, componentList []v1alpha1.Component) []updateTarget
+	GenerateRenovateConfigForNudge(target updateTarget, buildResult *BuildResult) (string, error)
+	CreateRenovaterPipeline(ctx context.Context, namespace string, targets []updateTarget, debug bool, buildResult *BuildResult) error
+}
+*/
+
+type DefaultComponentDependenciesUpdater struct {
```

Comment:
If we don't use any interface, word `Default` looks somewhat weird.

Reply:
changed